### PR TITLE
Purge gem cache directly in BundlerApi::Job

### PR DESCRIPTION
**Pull request by huacnlee**
Run example:

```
# of non yanked local gem versions: 720350
Finished Enqueuing Jobs!
Adding: repres-bootstrap-1.0.3
Purging /quick/Marshal.4.8/repres-bootstrap-1.0.3.gemspec.rz, /gems/repres-bootstrap-1.0.3.gem, /info/repres-bootstrap
Adding: repres-react-1.0.4
Purging /quick/Marshal.4.8/repres-react-1.0.4.gemspec.rz, /gems/repres-react-1.0.4.gem, /info/repres-react
```

### CODE REVIEW

File: lib/bundler_api/update/job.rb [hunk 1]
```diff
@@ -30,6 +31,7 @@ def run
     @mutex.synchronize do
       deps_added = insert_spec(spec)
       @gem_count.increment if @gem_count && (!deps_added.empty? || !@fix_deps)
+      @@cache.purge_gem(@payload)
```
Comment (segiddins) [inline lib/bundler_api/update/job.rb]: Please have this passed in as a parameter

**Comment by huacnlee** [comment]
@segiddins Just updated

File: Rakefile [hunk 2]
```diff
@@ -202,10 +203,12 @@ end
 
 desc "Add a specific single gem version to the database"
 task :add_spec, :name, :version, :platform, :prerelease do |t, args|
+  cache = BundlerApi::CacheInvalidator.new
+  mutex = Mutex.new
   args.with_defaults(:platform => 'ruby', :prerelease => false)
   payload = BundlerApi::GemHelper.new(args[:name], Gem::Version.new(args[:version]), args[:platform], args[:prerelease])
   database_connection do |db|
-    BundlerApi::Job.new(db, payload).run
+    BundlerApi::Job.new(db, payload, mutexm, 0, cache: cache).run
```
Comment (indirect) [inline Rakefile]: This is a typo (`mutexm`), but also why do you need to add a mutex to this call?

**Comment by huacnlee** [comment]
I have fixed typo, and remove unused option.

**Comment by indirect** [comment]
@homu r+

**Pull request merged**